"near operation" for graphql-codegen

richardgirges opened the pull request:
Closes #182

* `src/api/schema.generated.ts` now contains TS GQL schema with no operations
* `src/**/*.generated.tsx` are generated TS GQL operations that will live next to their counterpart `.graphql` files

CarsonF commented on codegen.yml:
```diff
@@ -15,11 +15,24 @@ generates:
     config:
       apolloClientVersion: 3
-  ./src/api/graphql.generated.tsx:
+  ./src/api/graphql.generated.ts:
```
```suggestion
  ./src/api/schema.generated.ts:
```

CarsonF commented on codegen.yml:
```diff
@@ -15,11 +15,24 @@ generates:
+  ./src/:
+    preset: near-operation-file
+    presetConfig:
+      extension: .generated.tsx
```
I don't think there's an JSX in these files, right? We could change it `.ts` and limit our eslint extensions.

CarsonF commented on src/api/fragments/location.generated.tsx:
```diff
@@ -0,0 +1,106 @@
+/* eslint-disable import/no-duplicates, @typescript-eslint/no-empty-interface */
```
This fragements folder may not make a lot of sense, esp now that less is in the api/ folder.
We don't need to change it now. But it would be good in the future to co-locate it with the `displayLocation` function and maybe move that somewhere else. At least a folder within api/ if not moving to components/.

richardgirges replied:
Noted, makes sense!

richardgirges commented:
@CarsonF updates pushed! Renamed all straggling `queries.graphql` files, changed `.generated.tsx` to `.generated.ts`, and also eliminated the `DevTest` scene as there is plenty of example code now.
